### PR TITLE
IRIS-386 java mobile capabilities and edit README.md

### PR DESCRIPTION
**Pull request by kb-kerem**

### CODE REVIEW

File: wd-java/src/test/java/com/example/TestUtils.java [hunk 1]
```diff
@@ -9,17 +12,50 @@
...
+        caps.setCapability("appium:automationName", "UiAutomator2");
+        return caps;
+    }
 
-        return new RemoteWebDriver(new URL(webDriverUrl), caps);
+    private static MutableCapabilities getAndroidCapabilities() {
+        MutableCapabilities caps = new MutableCapabilities();
+        caps.setCapability("platformName", "Android");
+        caps.setCapability("browserName", "Chrome");
+        caps.setCapability("appium:deviceName", "Google.*");
```
Comment (wswebcreation) [inline wd-java/src/test/java/com/example/TestUtils.java:54]: Please select a "specific" device, `Google.*` will return an available device that matches the regex, meaning a pixel 1-8

File: wd-java-testng/src/test/java/com/example/TestUtils.java [hunk 2]
```diff
@@ -1,25 +1,61 @@
...
 
 import java.net.MalformedURLException;
 import java.net.URL;
 
 public class TestUtils {
-    
+
     static RemoteWebDriver getWebDriver(String username, String accessKey) throws MalformedURLException {
+        // Set capabilities for WebDriver
+        DesiredCapabilities caps = new DesiredCapabilities();
```
Comment (aluedeke) [inline wd-java-testng/src/test/java/com/example/TestUtils.java:16]:
```suggestion
        var caps = new DesiredCapabilities();
```

File: wd-java-testng/src/test/java/com/example/TestUtils.java [hunk 3]
```diff
@@ -1,25 +1,61 @@
...
+        caps.setBrowserName("chrome");
+        return new RemoteWebDriver(getDriverUrl(username,accessKey), caps);
+    }
+
+    static AndroidDriver getAndroidEmulatorDriver(String username, String accessKey) throws MalformedURLException {
+        MutableCapabilities caps = getAndroidEmulatorCapabilities();
```
Comment (aluedeke) [inline wd-java-testng/src/test/java/com/example/TestUtils.java:22]:
```suggestion
        var caps = getAndroidEmulatorCapabilities();
```

File: wd-java-testng/src/test/java/com/example/TestUtils.java [hunk 4]
```diff
@@ -1,25 +1,61 @@
...
+        return new AndroidDriver(getDriverUrl(username, accessKey), caps);
+    }
+
+    static AndroidDriver getAndroidDriver(String username, String accessKey) throws MalformedURLException {
+        MutableCapabilities caps = getAndroidCapabilities();
```
Comment (aluedeke) [inline wd-java-testng/src/test/java/com/example/TestUtils.java:27]:
```suggestion
        var caps = getAndroidCapabilities();
```

File: wd-java-testng/src/test/java/com/example/TestUtils.java [hunk 5]
```diff
@@ -1,25 +1,61 @@
...
         // Can be found at "Driver creation" on https://app.saucelabs.com/user-settings
-        String webDriverUrl = "https://" + username + ":" + accessKey + "@ondemand.saucelabs.com:443/wd/hub";
+        return new URL("https://" + username + ":" + accessKey + "@ondemand.saucelabs.com:443/wd/hub");
+    }
 
-        // Set capabilities for WebDriver
-        DesiredCapabilities caps = new DesiredCapabilities();
-        caps.setBrowserName("chrome");
+    private static MutableCapabilities getAndroidEmulatorCapabilities() {
+        MutableCapabilities caps = new MutableCapabilities();
```
Comment (aluedeke) [inline wd-java-testng/src/test/java/com/example/TestUtils.java:41]: please change to var everywhere

File: wd-java/src/test/java/com/example/InventoryTest.java [hunk 6]
```diff
@@ -19,11 +19,14 @@ public class InventoryTest {
     private static final String accessKey = dotenv.get("SAUCE_ACCESS_KEY");
 
     private static VisualApi visual;
-    private static RemoteWebDriver driver;
+    private static RemoteWebDriver driver; // use AndroidDriver for Android
```
Comment (aluedeke) [inline wd-java/src/test/java/com/example/InventoryTest.java:22]: why do we need the AndroidDriver here?

File: wd-java-testng/pom.xml [hunk 7]
```diff
@@ -30,7 +30,13 @@
     <dependency>
       <groupId>org.seleniumhq.selenium</groupId>
       <artifactId>selenium-java</artifactId>
-      <version>4.10.0</version>
+      <version>4.14.1</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.appium</groupId>
```
Comment (aluedeke) [inline wd-java-testng/pom.xml:37]: Do the Android/iOS driver actually add any value here? Can't we just stick to the RemoteWebDriver?

**Review by aluedeke — COMMENTED** [review]
would be great if we could simplify more and use `var` keyword consistently

**Review by aluedeke — COMMENTED** [review]
Adding iOS would bre great as well.

File: wd-java-testng/pom.xml (hunk 7 above)
Comment (kb-kerem) [reply wd-java-testng/pom.xml]: Removed

File: wd-java/src/test/java/com/example/InventoryTest.java (hunk 6 above)
Comment (kb-kerem) [reply wd-java/src/test/java/com/example/InventoryTest.java]: Removed

File: wd-java-testng/src/test/java/com/example/TestUtils.java (hunk 5 above)
Comment (kb-kerem) [reply wd-java-testng/src/test/java/com/example/TestUtils.java]: Done

File: wd-java/src/test/java/com/example/TestUtils.java (hunk 1 above)
Comment (kb-kerem) [reply wd-java/src/test/java/com/example/TestUtils.java]: Done

**Comment by kb-kerem** [comment]
All is fixed and working now

**Review by aluedeke — APPROVED** [review]
maybe in the next iteration remove the redundant code between both examples

**Pull request merged**